sihl/src/gen_service_test.ml

```diff
@@ -0,0 +1,100 @@
+let template =
+  {|
+open Lwt.Syntax
+
+let testable_{{name}} =
+  Alcotest.testable {{module}}.pp (fun t1 t2 ->
+      String.equal t1.{{module}}.id t2.{{module}}.id)
+;;
+
+let create _ () =
+  let* () = Sihl.Cleaner.clean_all () in
+  let* () = {{module}}.clean () in
+  let* created = {{module}}.create {{create_values}} |> Lwt.map Result.get_ok in
+  let* (found : {{module}}.t) =
+    {{module}}.find created.{{module}}.id |> Lwt.map Option.get
+  in
+  Alcotest.(check testable_{{name}} "is same" found created);
+  Lwt.return ()
+;;
+
+let delete' _ () =
+  let* () = Sihl.Cleaner.clean_all () in
+  let* () = {{module}}.clean () in
+  let* created = {{module}}.create {{create_values}} |> Lwt.map Result.get_ok in
+  let* (found : {{module}}.t) =
+    {{module}}.find created.{{module}}.id |> Lwt.map Option.get
+  in
+  let* _ = {{module}}.delete found in
+  let* found =
+    {{module}}.find created.{{module}}.id
+  in
+  Alcotest.(check bool "was deleted" true (Option.is_none found));
+  Lwt.return ()
+;;
+
+let find_all _ () =
+  let* () = Sihl.Cleaner.clean_all () in
+  let* () = {{module}}.clean () in
+  let* created1 = {{module}}.create {{create_values}} |> Lwt.map Result.get_ok in
+  let* created2 = {{module}}.create {{create_values}} |> Lwt.map Result.get_ok in
+  let* ({{name}}s : {{module}}.t list) = {{module}}.query () in
+  Alcotest.(check (list testable_{{name}}) "has {{name}}s" [created2; created1] {{name}}s);
+  Lwt.return ()
+;;
+
+let suite =
+  Alcotest_lwt.
+    [ ( "crud {{name}}"
+      , [ test_case "create" `Quick create
+        ; test_case "delete" `Quick delete'
+        ; test_case "find_all" `Quick find_all
+        ] )
+    ]
+;;
+
+let services =
+  [ Sihl.Database.register (); Sihl.Database.Migration.PostgreSql.register () ]
+;;
+
+let () =
+  let open Lwt.Syntax in
+  Logs.set_level (Sihl.Log.get_log_level ());
+  Logs.set_reporter (Sihl.Log.cli_reporter ());
+  Lwt_main.run
+    (let* _ = Sihl.Container.start_services services in
```

Comment:
We should maybe consider imposing `let%lwt` on the user? Or is that a bad idea?

Reply:
(Considering we are imposing `let%html` on them)

Reply:
Good idea and that is planned, first we need to adjust `spin-sihl` and `sihl-demo` though.

Reply:
Okay!